### PR TITLE
fix: create endpoint check

### PR DESCRIPTION
**Pull request by elonniu**
## Description

- fixed create endpoint check

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: middleware_api/lambda/inference_v2/sagemaker_endpoint_api.py [hunk 1]
```diff
@@ -172,40 +172,40 @@ def sagemaker_endpoint_create_api(raw_event, ctx):
...
                 'statusCode': 400,
-                'errMsg': f"Creator {event.creator} has no permission to create Sagemaker",
+                'message': f"Creator {event.creator} has no permission to create Sagemaker",
             }
 
         endpoint_rows = ddb_service.scan(sagemaker_endpoint_table, filters=None)
         for endpoint_row in endpoint_rows:
             endpoint = EndpointDeploymentJob(**(ddb_service.deserialize(endpoint_row)))
-            if endpoint.endpoint_status == 'InService' and endpoint.status != 'deleted':
+            if endpoint.endpoint_status != 'Deleted' and endpoint.status != 'deleted':
```
Comment (yike5460) [inline middleware_api/lambda/inference_v2/sagemaker_endpoint_api.py:187]: we should not use string for the state comparison, use a mapping array instead. e.g.
```
class PermissionStatus:
    DELETE = 'Delete'
```
also we should add document to elaborate the matrix of role, permission and corresponding action

File: middleware_api/lambda/inference_v2/sagemaker_endpoint_api.py [hunk 2]
```diff
@@ -172,40 +172,40 @@ def sagemaker_endpoint_create_api(raw_event, ctx):
...
                 for role in event.assign_to_roles:
                     if role in endpoint.owner_group_or_role:
                         return {
                             'statusCode': 400,
-                            'errMsg': f"role [{role}] has a valid endpoint already, not allow to have another one",
+                            'message': f"role [{role}] has a valid endpoint already, not allow to have another one",
```
Comment (yike5460) [inline middleware_api/lambda/inference_v2/sagemaker_endpoint_api.py:192]: user a generic class for the message returned, e.g.

```
class HttpStatusCode:
    OK = 200
    NoContent = 204
    BadRequest = 400
    NotFound = 404
    InternalServerError = 500

# Mapping status codes to descriptions
http_status_descriptions = {
    HttpStatusCode.OK: "OK",
    HttpStatusCode.NoContent: "No Content",
    HttpStatusCode.BadRequest: "Bad Request",
    HttpStatusCode.NotFound: "Not Found",
    HttpStatusCode.InternalServerError: "Internal Server Error"
}

class StatusCode:
    def __init__(self, code):
        self.code = code
        self.description = http_status_descriptions.get(code, "Unknown Status Code")

    def __str__(self):
        return f"Status Code: {self.code} - {self.description}"
```

File: middleware_api/lambda/inference_v2/sagemaker_endpoint_api.py [hunk 3]
```diff
@@ -172,40 +172,40 @@ def sagemaker_endpoint_create_api(raw_event, ctx):
...
                         }
 
+        _create_sagemaker_model(sagemaker_model_name, image_url, model_data_url)
+
+        _create_sagemaker_endpoint_config(sagemaker_endpoint_config, s3_output_path, sagemaker_model_name,
```
Comment (yike5460) [inline middleware_api/lambda/inference_v2/sagemaker_endpoint_api.py:197]: make sure those 2 APIs don't have any dependencies and is irrespective of order

File: middleware_api/lambda/inference_v2/sagemaker_endpoint_api.py [hunk 4]
```diff
@@ -172,40 +172,40 @@ def sagemaker_endpoint_create_api(raw_event, ctx):
...
+                                          initial_instance_count, instance_type)
+
+        logger.info('Creating new model endpoint...')
+        response = sagemaker.create_endpoint(
+            EndpointName=sagemaker_endpoint_name,
+            EndpointConfigName=sagemaker_endpoint_config
+        )
+        logger.info(f"Successfully created endpoint: {response}")
+
+        current_time = str(datetime.now())
```
Comment (yike5460) [inline middleware_api/lambda/inference_v2/sagemaker_endpoint_api.py:207]: in which line we use such time?

**Review by yike5460 — COMMENTED** [review]
comments leaved

File: middleware_api/lambda/inference_v2/sagemaker_endpoint_api.py (hunk 4 above)
Comment (elonniu) [reply middleware_api/lambda/inference_v2/sagemaker_endpoint_api.py:208]: in 213 line, just record start time

**Review by yike5460 — APPROVED** [review]
lgtm

**Pull request merged**